src/Pim/Component/Catalog/Factory/ProductValue/MediaProductValueFactory.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Pim\Component\Catalog\Factory\ProductValue;
+
+use Akeneo\Component\FileStorage\Model\FileInfoInterface;
+use Akeneo\Component\FileStorage\Repository\FileInfoRepositoryInterface;
+use Akeneo\Component\StorageUtils\Exception\InvalidPropertyException;
+use Pim\Component\Catalog\Exception\InvalidArgumentException;
+use Pim\Component\Catalog\Model\AttributeInterface;
+
+/**
+ * Factory that creates media product values.
+ *
+ * @internal  Please, do not use this class directly. You must use \Pim\Component\Catalog\Factory\ProductValueFactory.
+ *
+ * @author    Damien Carcel (damien.carcel@akeneo.com)
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class MediaProductValueFactory implements ProductValueFactoryInterface
+{
+    protected $fileInfoRepository;
+
+    /** @var string */
+    protected $mediaProductValueClass;
+
+    /** @var string */
+    protected $supportedAttributeType;
+
+    /**
+     * @param FileInfoRepositoryInterface $fileInfoRepository
+     * @param string                      $mediaProductValueClass
+     * @param string                      $supportedAttributeType
+     */
+    public function __construct(
+        FileInfoRepositoryInterface $fileInfoRepository,
+        $mediaProductValueClass,
+        $supportedAttributeType
+    ) {
+        if (!class_exists($mediaProductValueClass)) {
```

Comment:
Should we throw exception in constructor? If an exception is thrown there it should be a logic exception.